WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -90,24 +90,31 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 					trackedWallets.Remove(finishedCoinJoin.Wallet.WalletName);
 					finishedCoinJoin.CancellationTokenSource.Dispose();
 
-					var finishedCoinJoinTask = finishedCoinJoin.CoinJoinTask;
-					if (finishedCoinJoinTask.IsCompletedSuccessfully)
+					try
 					{
-						Logger.LogInfo("Coinjoin client finished successfully!");
-					}
-					else if (finishedCoinJoinTask.IsFaulted)
-					{
-						if (finishedCoinJoinTask.Exception?.InnerException is InvalidOperationException ioe)
+						var success = await finishedCoinJoin.CoinJoinTask.ConfigureAwait(false);
+						if (success)
+						{
+							Logger.LogInfo("Coinjoin client finished successfully!");
+						}
+						else
 						{
-							await Task.Delay(TimeSpan.FromSeconds(5), stoppingToken).ConfigureAwait(false);
-							continue;
+							Logger.LogInfo("Coinjoin client finished with error. Transaction not broadcasted.");
 						}
-						Logger.LogError(finishedCoinJoinTask.Exception!);
 					}
-					else if (finishedCoinJoinTask.IsCanceled)
+					catch (InvalidOperationException ioe)
+					{
+						Logger.LogError(ioe);
+						await Task.Delay(TimeSpan.FromSeconds(5), stoppingToken).ConfigureAwait(false);
+					}
+					catch (OperationCanceledException)
 					{
 						Logger.LogInfo("Coinjoin client was cancelled.");
 					}
+					catch (Exception e)
+					{
+						Logger.LogError(e);
```

Comment:
It's a bit pity that `WalletTrackingData` does not have some kind of unique identifier to log it here.

Reply:
Ummm yes, i think an identifier wouldn't help much but without any kind of reference this log entry would be pointless too. I will add the wallet name at least.